src/api/api.js

```diff
@@ -19,441 +53,322 @@ export const getMessageConversations = ({
-                        instance.currentUser.authorities
-                    )
-                    return configuration.then(configurationResult => ({
-                        authorized:
-                            userAuthorities.includes('ALL') ||
-                            !!result.userGroups.find(
-                                group =>
-                                    group.id ===
-                                    configurationResult.feedbackRecipients.id
-                            ),
-                        feedbackRecipientsId:
-                            configurationResult.feedbackRecipients.id,
-                    }))
-                })
-                .catch(error => {
-                    throw error
-                })
-        )
-        .catch(error => {
-            throw error
-        })
-
-/* Recipient search */
-const MAX_RECIPIENT = 10
-const searchOrganisationUnits = searchValue =>
-    getD2Instance()
-        .then(instance =>
-            instance.Api.getApi().get('organisationUnits', {
-                fields: 'id, displayName',
-                pageSize: MAX_RECIPIENT,
-                filter: [`displayName:token:${searchValue}`, 'users:gte:1'],
-            })
-        )
-        .then(result => result)
-        .catch(error => {
-            throw error
-        })
-
-const searchUserGroups = searchValue =>
-    getD2Instance()
-        .then(instance =>
-            instance.Api.getApi().get('userGroups', {
-                fields: 'id, displayName',
-                pageSize: MAX_RECIPIENT,
-                filter: [`displayName:token:${searchValue}`],
-            })
-        )
-        .then(result => result)
-        .catch(error => {
-            throw error
-        })
-
-export const searchRecipients = ({
+    await engine.mutate({
+        resource: 'messageConversations',
+        type: 'create',
+        data: {
+            subject,
+            users,
+            userGroups,
+            organisationUnits,
+            text,
+            attachments,
+        },
+    })
+
+export const sendFeedbackMessage = async (subject, text) =>
+    await engine.mutate({
+        resource: 'messageConversations/feedback',
+        type: 'create',
+        params: { subject },
+        data: text,
+    })
+
+export const replyMessage = async ({
+    message,
+    internalReply,
+    attachments,
+    id,
+}) => {
+    const params = { internal: internalReply }
+
+    if (attachments.length > 0) {
+        params.attachments = attachments
+    }
+
+    return await engine.mutate({
+        resource: `messageConversations/${id}`,
+        type: 'create',
+        params,
+        data: message,
+    })
+}
+
+export const deleteMessageConversation = async (
+    messageConversationId,
+    currentUserId
+) =>
+    await engine.mutate({
+        resource: `messageConversations/${messageConversationId}/${currentUserId}`,
+        type: 'delete',
+    })
```

Comment:
Would it not be equivalent to omit the async/await and just return `engine.mutate` directly here? If so, I'd prefer that.

Reply:
(Goes for the other areas that use this pattern as well)

Reply:
👍

Reply:
Solved in 7f16890b41e154fe70e1b1f9fa927b0adb79c258